### ISSUE TITLE
Issues related to top bar appearing in the chat

### ISSUE THREAD

**Issue by VolodLytvynenko** [status-im/status-react#16160]
introduced by https://github.com/status-im/status-mobile/pull/15204
### Steps to reproduce:
1. Open a chat with only a few messages.
2. Start typing and sending messages in the chat. (check the tap bar during each message sending)

### Actual result:
As the user types and sends messages, the top bar of the chat screen does not appear and only the 'back' button is shown instead

https://github.com/status-im/status-mobile/assets/52490791/85c56cd0-b0f1-44f3-87dd-c32809fcdb38

### Expected result:
When the user starts typing and sending messages in a chat with only a few messages, the top bar of the chat screen should appear

### Additional info:
Sometimes the top bar does not appear even after reopening the chat if there are only a few messages in the chat. This behavior has been observed when there are approximately 10 messages in the chat.

### Device:
iPhone 11 pro max, IOS 16

**Comment by churik** [comment]
сс @alwx

**Comment by VolodLytvynenko** [comment]
## ISSUE 2: [ANDROID] Premature appearance of the top bar in the chat when the composer is focused


## Steps to reproduce:
1. Open a chat without messages or with a few messages (user details page should be visible if a few messages are in chat)
2. Tap on the composer

### Actual result:
As soon as the composer is focused for typing, the top bar in the chat screen appears 

https://github.com/status-im/status-mobile/assets/52490791/864e86a5-65c8-4ef5-af37-16a646041e3c



### Expected result:
The top bar in the chat screen should appear only when "user details page UI" is already hidden

**Comment by Parveshdhull** [comment]
Please only address these issues, once https://github.com/status-im/status-mobile/pull/15893 is merged.

**Comment by Parveshdhull** [comment]
Creating conflicts, and issues might return (don't know)

https://github.com/status-im/status-mobile/pull/15893#issuecomment-1576567809

**Comment by Parveshdhull** [comment]
> Please only address these issues, once #15893 is merged.

Never mind, without fixing this issue, PR can't even be tested. Back button etc. gets hidden behind status-bar

**Comment by alwx** [comment]
Partially fixed here: https://github.com/status-im/status-mobile/pull/16148 (at least the part which blocks you)

**Comment by pavloburykh** [comment]
> Partially fixed here: https://github.com/status-im/status-mobile/pull/16148 (at least the part which blocks you)

@alwx hi! Could you please specify, which of the ISSUES mentioned in this ticket has been fixed in https://github.com/status-im/status-mobile/pull/16148? 

As far as I see both issues are still there. 

I see that you linked another issue https://github.com/status-im/status-mobile/issues/16141 to https://github.com/status-im/status-mobile/pull/16148. 

Should we also link current issue to https://github.com/status-im/status-mobile/pull/16148 or it is out of scope of that PR?

**Comment by pavloburykh** [comment]
@Parveshdhull hi! Which of the issues blocks you in context of https://github.com/status-im/status-mobile/pull/15893: those that are mentioned above in this ticket or this one https://github.com/status-im/status-mobile/issues/16141 ?

**Comment by Parveshdhull** [comment]
hi @pavloburykh, Issues mentioned in #16141

**Comment by alwx** [comment]
It's becoming a bit of a mess — I kinda lost track of these top bar issues, most do actually duplicate each other. What's blocking is this one: https://github.com/status-im/status-mobile/issues/16141 because of not being able to press the back button but that's fixed in this PR: https://github.com/status-im/status-mobile/pull/16148

**Comment by pavloburykh** [comment]
> It's becoming a bit of a mess — I kinda lost track of these top bar issues, most do actually duplicate each other. What's blocking is this one: #16141 because of not being able to press the back button but that's fixed in this PR: #16148

@alwx You can search for top bar related issues by key expression "introduced by #15204 ". If you consider any of issues to be a duplicate or already adressed  - you are good to close such issues. 

https://github.com/status-im/status-mobile/issues?q=is%3Aissue+is%3Aopen+introduced+by+%2315204

<img width="1274" alt="Issues · status-im:status-mobile 2023-06-06 13-20-20" src="https://github.com/status-im/status-mobile/assets/97245802/8e064f2b-fb58-4eb3-994a-aa8cededb34a">

**Comment by alwx** [comment]
From #16266:

1) when opening chat the top bar behaves wrong


https://github.com/status-im/status-mobile/assets/11790366/d8346666-0bd0-4069-b5b0-d3e8234b24a8



2) if the chat screen opened in some state it might look broken

<img width="330" alt="image" src="https://github.com/status-im/status-mobile/assets/11790366/89d26852-5d6b-4abd-bfaf-2775e364e567">

**Comment by pavloburykh** [comment]
> ISSUE 1: [IOS] Top bar does not appear when typing and sending messages in a chat initially with only a few messages

Some more details regarding this issue: 

1. Bug is happening not only on IOS but also on Android. 
2. Bug is reproducible not only with few messages in chat history but also with long history. Sometimes top bar just glitches after chat opening as shown [here](https://github.com/status-im/status-mobile/issues/16160#issuecomment-1591347003) but sometimes it is not rendered at all. See videos below. 

https://github.com/status-im/status-mobile/assets/97245802/59cb385e-6a37-4631-98fb-08330d922d42


https://github.com/status-im/status-mobile/assets/97245802/573fb7a1-90ec-4bf6-8f68-333d3af66d6d

**Comment by qoqobolo** [comment]
Adding to remember to check the pinned messages banner when there is little chat history:

> ISSUE 1: [IOS] Top bar does not appear when typing and sending messages in a chat initially with only a few messages

<img width="328" alt="Screenshot 2023-06-20 at 16 01 28" src="https://github.com/status-im/status-mobile/assets/67952253/ddb84804-2959-45fd-b66f-862e8328809e">

**Comment by flexsurfer** [comment]
pinned messages banner won't disappear on scroll when you add the first pinned message, after reopening the chat it works as expected

**Comment by pavloburykh** [comment]
### ISSUE Profile picture is not displayed in header when user is at the beginning of chat history

![photo_2023-06-22 18 35 00](https://github.com/status-im/status-mobile/assets/97245802/f5157207-d476-44d0-9f7e-3d70856351b5)

https://github.com/status-im/status-mobile/assets/97245802/5698ec7d-6216-4500-9330-6aae7bc81240

**Comment by pavloburykh** [comment]
> pinned messages banner won't disappear on scroll when you add the first pinned message, after reopening the chat it works as expected

@alwx here is demonstration of such behaviour


https://github.com/status-im/status-mobile/assets/97245802/782adfad-a6cb-4c63-aaad-47b95512d1c5

![photo_2023-06-23 13 55 45](https://github.com/status-im/status-mobile/assets/97245802/47004dbc-fee1-46d8-9988-e80d337ecdf9)
